### DISCUSSION TITLE
Any interest in using a shared package for UUIDs?

### DISCUSSION THREAD

**Discussion by koenvervloesem** [hbldh/bleak#861] [Ideas]
For another project ([Theengs Explorer](https://github.com/theengs/explorer)) using Bleak I needed an up-to-date list of company IDs. I created the project [bluetooth-numbers](https://github.com/koenvervloesem/bluetooth-numbers) I intend to keep synchronized with Nordic Semiconductor's [Bluetooth Numbers Database](https://github.com/NordicSemiconductor/bluetooth-numbers-database), having Company IDs, Service UUIDs, Characteristic UUIDs and Descriptor UUIDs. The goal is to provide a shared resource so various Python projects that deal with Bluetooth don't have to replicate this effort.

Is there any interest from Bleak in using the bluetooth-numbers package so you don't need to keep the UUIDs updated yourselves? Some examples of its usage:

Get the description of a company ID:

```python
>>> from bluetooth_numbers.companies import company
>>> company[0x0499]
'Ruuvi Innovations Ltd.'
```

Get the description of a service UUID:

```python
>>> from bluetooth_numbers.services import service
>>> from uuid import UUID
>>> service[0x180F]
'Battery Service'
>>> service[UUID("6E400001-B5A3-F393-E0A9-E50E24DCCA9E")]
'Nordic UART Service'
```

Get the description of a characteristic UUID:

```python
>>> from bluetooth_numbers.characteristics import characteristic
>>> from uuid import UUID
>>> characteristic[0x2A37]
'Heart Rate Measurement'
>>> characteristic[UUID("6E400002-B5A3-F393-E0A9-E50E24DCCA9E")]
'UART RX Characteristic'
```

Get the description of a descriptor UUID:

```python
>>> from bluetooth_numbers.descriptors import descriptor
>>> descriptor[0x2901]
'Characteristic User Descriptor'
```
The project is brand new with only one other project using it for the moment, so if you have remarks about the API before using it in Bleak, I can definitely change it.

I intend to periodically publish new releases of the package with updated data from the Bluetooth Numbers Database.

**Comment by dlech** [comment]
Hi @koenvervloesem, this looks great! It's something we could definitely use.

It would also be useful if the bluetooth-numbers package provided a reverse lookup. Enums might be a nice way to do this. For the services/characteristics/descriptors, the "identifier" could be used as the enum member. For example "org.bluetooth.service.battery_service" would become:

```python
class Service(Enum):
    ...
    ORG_BLUETOOTH_BATTERY_SERVICE = 0x180F
    ...
```

Company names would be a bit messy though since they don't have an identifier.
